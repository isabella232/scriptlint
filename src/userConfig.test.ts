import loadConfig, {sanitizeConfig, defaultConfig} from "./userConfig";

const validConfig = {
	extends: ["foo"],
	rules: {foo: "bar"},
	ignoreScripts: ["foo"],
};

const invalidConfig = {
	invalid: 3,
};

describe("reporter.ts", () => {
	it("should sanitize configs: empty config => default", () => {
		expect(sanitizeConfig({})).toEqual(defaultConfig);
	});
	it("should sanitize configs: null => default", () => {
		expect(sanitizeConfig(null)).toEqual(defaultConfig);
	});
	it("should sanitize configs: invalid keys", () => {
		expect(sanitizeConfig(invalidConfig)).toEqual(defaultConfig);
	});
	it("should sanitize configs: valid => valid", () => {
		expect(sanitizeConfig(validConfig)).toEqual(validConfig);
	});
	it("should sanitize configs: extends not empty", () => {
		expect(sanitizeConfig({
			extends: [],
			rules: {foo: "bar"},
		})).toEqual({
			...defaultConfig,
			rules: {foo: "bar"},
		});
	});

	test("loadConfig()", () => {
		const loaded = loadConfig();

		expect(loaded).toMatchSnapshot();
	});

	test("loadConfig() with config missing", () => {
		const loaded = loadConfig("missing");

		expect(loaded).toBe(defaultConfig);
	});
});

export {};
