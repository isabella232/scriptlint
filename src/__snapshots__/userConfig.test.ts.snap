// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reporter.ts loadConfig() 1`] = `
Object {
  "extends": Array [
    "scriptlint/strict",
  ],
  "ignoreScripts": Array [
    "foobar",
  ],
  "rules": Object {
    "mandatory-dev": false,
  },
}
`;
